__tests__/integration/send-tx.test.js

```diff
@@ -0,0 +1,1005 @@
+    // 800 remained on Wallet1
+    done();
+  });
+
+  it('should send a custom token with multiple inputs', async done => {
+    const sendOptions = {
+      inputs: [
+        {
+          hash: tkaTx1.hash,
+          index: 1
+        }, // Using index 1 of main transaction
+        tkaTx2 // Change on wallet 1
+      ],
+      outputs: [{
+        address: await wallet2.getAddressAt(0),
+        value: 1800
+      }],
+      token: tokenA,
+    };
+    const tx = await wallet1.sendTx({
+      fullObject: sendOptions
+    });
+
+    expect(tx.success).toBe(true);
+    expect(tx.hash).toBeDefined();
+
+    // All 2000 TKA are now on Wallet2
+    done();
+  });
+
+  it('should send a custom token', async done => {
+    // Sending all TokenA back to wallet 1, address 0
+    const sendOptions = {
+      outputs: [{
+        address: await wallet1.getAddressAt(0),
+        value: 2000
+      }],
+      token: tokenA,
+    };
+
+    const tx = await wallet2.sendTx({
+      fullObject: sendOptions
+    });
+
+    expect(tx.success).toBe(true);
+    expect(tx.hash).toBeDefined();
+    done();
+  });
+
+  it('should send a custom token with multi inputs and outputs', async done => {
+    // Spreading Token A into three output addresses on wallet2
+    const spreadTxOptions = {
+      outputs: [
+        {
+          address: await wallet2.getAddressAt(2),
+          value: 900
+        },
+        {
+          address: await wallet2.getAddressAt(3),
+          value: 800
+        },
+        {
+          address: await wallet2.getAddressAt(4),
+          value: 300
+        }
+      ],
+      token: tokenA,
+    };
+    const spreadTx = await wallet1.sendTx({
+      fullObject: spreadTxOptions
+    });
+
+    // Consolidating these funds back into wallet1 in two addresses
+    const consolidateTxOptions = {
+      outputs: [
+        {
+          address: await wallet1.getAddressAt(3),
+          value: 1600
+        },
+        {
+          address: await wallet1.getAddressAt(4),
+          value: 400
+        }
+      ],
+      inputs: [
+        {
+          hash: spreadTx.hash,
+          index: 0
+        },
+        {
+          hash: spreadTx.hash,
+          index: 1
+        },
+        {
+          hash: spreadTx.hash,
+          index: 2
+        }
+      ],
+      token: tokenA
+    };
+    const consolidateTx = await wallet2.sendTx({
+      fullObject: consolidateTxOptions
+    });
+
+    expect(consolidateTx.success).toBe(true);
+    expect(consolidateTx.hash).toBeDefined();
+    done();
+  });
+
+  it('should send a multi-input, multi token transaction', async done => {
+    const tx = await wallet3.sendTx({
+      fullObject: {
+        inputs: [
+          {
+            hash: tokenB.uid,
+            index: 0,
+          },
+          {
+            hash: tokenB.uid,
+            index: 1
+          },
+        ],
+        outputs: [
+          {
+            address: await wallet3.getAddressAt(7),
+            value: 1000,
```

Comment:
Is it the full amount of utxos? We could try to use less than the full amount, this test will be a really good one. We will use selected utxos for tokenB, the lib will select utxos for HTR, then we will have change outputs for tokenB and maybe for HTR, will be really good

Reply:
A complex test was implemented on [1846a50](https://github.com/HathorNetwork/hathor-wallet-headless/pull/152/commits/1846a5009722e29a4965e8f5909cfb91e0ba6d90) for this purpose.

Can you check if it covers what you have thought of here?